src/components/FileEditorContainer.tsx

```diff
@@ -23,7 +23,7 @@ const FileEditorContainer: React.FC<FileEditorContainerProps> = () => {
   const onFileSelect = async (path: string) => {
     if (selectedFilePath && editorContent !== lastSavedContentRef.current) {
       try {
-        await window.files.writeFile(selectedFilePath, editorContent); // save the current content.
+        await window.files.writeFile(selectedFilePath, editorContent, true); // save the current content.
```

Comment:
Is there a close-window event? There is the scenario where the user closes the window before selecting another file too, I think?

This seems like it is saving on selecting another file, which makes total sense too of course. But now that we don't have save being called every other second, closing a window might cause user to lose the current note working on?

Reply:
Actually this concept of indexing on boot feels like something that we can optimize as well. Perhaps its not an issue with low notes count situations, but I would prefer for us to index on window close because that is limited to the one file the user was last editing.

As opposed to a full reindexing each time we boot afresh, its my main gripe with notion tbh. Hate the slow bootup on mobile app

Reply:
It doesn't do a full index on boot. It checks which files need indexing and only indexes those. This is kind of designed so that the application overall can maintain less state and just be opened in any directory, or if say the user has added more files since last boot through another app. 

But yes there is room to optimise this whole flow in different ways - I suspect there may be issues around RAM for large corpuses given that the whole vector db is loaded into memory to check which files are missing. One idea could be to just do this quietly in the background...